protocol/x/clob/e2e/permissioned_keys_test.go

```diff
@@ -0,0 +1,561 @@
+package clob_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	testtx "github.com/dydxprotocol/v4-chain/protocol/testutil/tx"
+	aptypes "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	"github.com/stretchr/testify/require"
+)
+
+type TestBlockWithMsgs struct {
+	Block uint32
+	Msgs  []TestSdkMsg
+}
+
+type TestSdkMsg struct {
+	Msg            sdk.Msg
+	Authenticators []uint64
+	Fees           sdk.Coins
+	Gas            uint64
+	AccountNum     []uint64
+	SeqNum         []uint64
+	Signers        []cryptotypes.PrivKey
+
+	ExpectedRespCode uint32
+	ExpectedLog      string
```

Comment:
Could we add an `ExpectedGas` field?

Reply:
hmm these are short term orders with infinite gas meters and also IIUC if a transaction fails check tx, it doesn't incur any gas?

I can add expected gas for success cases, for stateful orders

Reply:
Oh I was referring to failed transactions that are not ST orders (e.g. transfer/send) since those should still incur gas cost even if transaction failed.